data/Builder.go

```diff
@@ -0,0 +1,82 @@
+package data
+
+import (
+	"fmt"
+
+	"github.com/tidepool-org/platform/validate"
+)
+
+const (
+	InvalidTypeTitle = "Invalid type"
+	InvalidDataTitle = "Invalid data"
+)
+
+type Datum map[string]interface{}
+type DatumArray []Datum
+
+type BuiltDatum interface{}
```

Comment:
Are we going to have to use `BuiltDatum` all over the place. I think I would change `Datum` to `RawDatum` and `BuiltDatum` to `Datum`.  More clear.  Since it is a global change, I can do it if you want (with a regex find/replace).

Or, we can keep it as-is.